### PR TITLE
New team wizard: support subteams

### PR DESCRIPTION
**Pull request by jakob223**
This ticket was for the last screen in the subteam wizard, but it was hard to test that without integrating the rest of it too.

It doesn't submit yet because we're waiting on an RPC for that, but otherwise the new subteam wizard should be fully working.

cc @keybase/y2ksquad

### CODE REVIEW

**Comment by jakob223** [comment]
bump @keybase/react-hackers

File: shared/teams/new-team/wizard/add-subteam-members.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  const doneLabel = selectedMembers.size ? 'Done' : 'Skip'
+
+  const renderItem = (_: number, m: Types.MemberInfo) => (
+    <Kb.ListItem2
+      type="Small"
+      icon={<Kb.Avatar username={m.username} size={32} />}
+      body={
+        <Kb.Box2 direction="vertical">
+          <Kb.ConnectedUsernames type="BodySemibold" usernames={[m.username]} />
+          <Kb.Text type="BodySmall">{m.fullName}</Kb.Text>
```
Comment (buoyad) [inline shared/teams/new-team/wizard/add-subteam-members.tsx]: `lineClamp`

File: shared/teams/new-team/wizard/add-subteam-members.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          <Kb.SearchFilter
+            size="full-width"
+            onChange={setFilter}
+            value={filter}
+            placeholderText={`Search ${parentMembers.length} members`}
+            icon="iconfont-search"
+            style={styles.search}
+          />
+        </Kb.Box2>
+        {/* TODO: once it's easier to make a single different-height header, make this part of the list2 */}
```
Comment (buoyad) [inline shared/teams/new-team/wizard/add-subteam-members.tsx]: I don't think it's bad this doesn't scroll. But agree would be nice for this to be easier.

File: shared/teams/new-team/wizard/new-team-info.tsx [hunk 3]
```diff
@@ -93,25 +118,43 @@ const NewTeamInfo = () => {
...
-          maxLength={16}
-          autoFocus={true}
-        />
+      <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.body} gap="tiny">
+        {parentName ? (
+          <Kb.NewInput
+            autoFocus={true}
+            maxLength={16}
+            onChangeText={setName}
+            prefix={`${parentName}.`}
```
Comment (buoyad) [inline shared/teams/new-team/wizard/new-team-info.tsx]: Alignment of this is off on mobile, I wonder if something changed / if it's broken everywhere. I remember it working when implemented. :(
image

File: shared/teams/new-team/wizard/new-team-info.tsx [hunk 4]
```diff
@@ -151,9 +194,13 @@ const NewTeamInfo = () => {
           checked={openTeam}
           onCheck={v => (rolePickerIsOpen ? undefined : setOpenTeam(v))}
         />
+        {teamWizardState.teamType === 'subteam' && (
+          <Kb.Checkbox onCheck={setAddYourself} checked={addYourself} label="Add yourself to the team" />
```
Comment (buoyad) [inline shared/teams/new-team/wizard/new-team-info.tsx]: as per discussion could be nice to add `LayoutAnimation` to the open team dropdown. with this checkbox added the extra empty space is apparent.

File: shared/teams/new-team/wizard/add-subteam-members.tsx (hunk 2 above)
Comment (jakob223) [reply shared/teams/new-team/wizard/add-subteam-members.tsx]: I asked Cècile and she said it should scroll but I'd already implemented it this way by the time she replied so I added this comment.

File: shared/teams/new-team/wizard/new-team-info.tsx [hunk 5]
```diff
@@ -166,11 +172,17 @@ const NewTeamInfo = () => {
...
+              {(!Styles.isMobile || openTeam) && (
+                <Kb.Box2
+                  direction="horizontal"
+                  gap="xtiny"
+                  alignSelf="flex-start"
+                  alignItems="center"
+                  className={Styles.classNames('hideableDropdown', {hidden: !openTeam})}
+                >
                   <Kb.Text type="BodySmall">People will join as</Kb.Text>
                   <FloatingRolePicker
```
Comment (buoyad) [inline shared/teams/new-team/wizard/new-team-info.tsx]: Owners is currently allowed in the dropdown but I think reader,writer,admin are the only valid options for this

File: shared/teams/new-team/wizard/new-team-info.tsx [hunk 6]
```diff
@@ -65,9 +65,15 @@ const NewTeamInfo = () => {
   }, [teamname, name.length, setTeamNameTaken, checkTeamNameTaken, setTeamNameTakenStatus])
```
Comment (buoyad) [inline shared/teams/new-team/wizard/new-team-info.tsx]: This page is missing a cancel link on mobile.

File: shared/teams/new-team/wizard/new-team-info.tsx [hunk 7]
```diff
@@ -65,9 +65,15 @@ const NewTeamInfo = () => {
...
 
   const [description, setDescription] = React.useState(teamWizardState.description)
-  const [openTeam, setOpenTeam] = React.useState(
+  const [openTeam, _setOpenTeam] = React.useState(
     teamWizardState.name ? teamWizardState.open : teamWizardState.teamType === 'community'
   )
+  const setOpenTeam = () => {
+    if (Styles.isMobile) {
+      Kb.LayoutAnimation.configureNext(Kb.LayoutAnimation.Presets.easeInEaseOut)
```
Comment (buoyad) [inline shared/teams/new-team/wizard/new-team-info.tsx]: :+1:

**Pull request merged**